pkg/proxy/proxy/transparent.go

```diff
@@ -115,31 +120,82 @@ func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate,
 	}
 }
 
+func (t TransparentProxyService) Shutdown(ctx context.Context) error {
+	if t.adminShutdown != nil {
+		if err := t.adminShutdown(ctx); err != nil {
+			return err
+		}
+	}
+	if t.httpShutdown != nil {
+		if err := t.httpShutdown(ctx); err != nil {
+			return err
+		}
+	}
+	if t.tlsShutdown != nil {
+		if err := t.tlsShutdown(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 // ProxyHTTP serves an endpoint that transparently redirects HTTP connections to the proxy server.
 // This endpoint also explicitly (i.e. non-transparently) proxies HTTP and TLS connections.
-func (t TransparentProxyService) ProxyHTTP(addr string) {
+func (t *TransparentProxyService) ProxyHTTP(addr string) {
```

Comment:
hmm actually it comes to mind that multiple calls to this function would fail to tear down the prior versions as it would override the shutdown func member. you could make it a slice and append entries instead

Reply:
Created a slice and just had all the shutdown functions appended to this